File: 2126/Solution.go
func asteroidsDestroyed(mass int, asteroids []int) bool {
    max := 0
    for _, val := range asteroids {
        if max < val {
            max = val
        }
    }
    a := make([]int, max + 1)
    for _, val := range asteroids {
        a[val]++
    }
    for i := range max + 1 {
        if i > mass {
            return false
        }
        if mass > 100000 {
            return true
        }
        mass += a[i] * i
    }
    return true
}

File: 2126/Solution.java
class Solution {
    public boolean asteroidsDestroyed(int mass, int[] asteroids) {
        int max = 0;
        for(int i : asteroids) if(max < i) max = i;
        int[] map = new int[max+1];
        for(int i : asteroids) map[i]++;
        long sum = mass;
        for(int i = 0; i <= max; i++){
            if(i > sum) return false;
            if(sum >= 100000) return true;
            sum += i * map[i];
        }
        return true;
    }
}
